src/pkg/utils/sbom.go

```diff
@@ -47,23 +47,26 @@ func SBOMExtractor(dst string, SBOMArtifactPathMap map[string]string) func(ctx c
 		if err != nil {
 			return err
 		}
-		buffer := make([]byte, f.Size()-1)
-		_, err = open.Read(buffer)
-		if err != nil {
-			return err
-		}
-		err = open.Close()
-		if err != nil {
-			return err
-		}
-		path := filepath.Join(dst, config.BundleSBOM, f.NameInArchive)
-		// todo: handle collisions? especially for zarf-component SBOM files?
-		err = os.WriteFile(path, buffer, 0644)
-		if err != nil {
-			return err
+		size := f.Size() - 1
+		if size > 0 {
+			buffer := make([]byte, size)
+			_, err = open.Read(buffer)
+			if err != nil {
+				return err
+			}
+			err = open.Close()
+			if err != nil {
+				return err
+			}
+			path := filepath.Join(dst, config.BundleSBOM, f.NameInArchive)
+			// todo: handle collisions? especially for zarf-component SBOM files?
```

Comment:
if you more simply wrap the tarballs from each package that will naturally separate them and may simplify the logic for matching OCI descriptors and extracting things from them.

Also note how `zarf package inspect <package> --sbom-out directory` works (it makes sub dirs for the package by name).

Reply:
This logic also clobbers the compare.html which will cause weirdness in the SBOM selector when going to the compare screen.

Reply:
(it may be good to have a tech debt issue / discussion on how the Zarf team can help - it may be good for y'all to make your own high level compare file and such too).

Reply:
Thanks @Racer159 , created the following issue to address the above: https://github.com/defenseunicorns/uds-cli/issues/305